ProjFS.Mac/PrjFSKext/AssociativeArrayTests/AssociativeArrayTests.mm

```diff
@@ -0,0 +1,135 @@
+#import <XCTest/XCTest.h>
+#include <list>
+#import "../PrjFSKext/AssociativeArray.hpp"
+
+@interface AssociativeArrayTests : XCTestCase
+
+@end
+
+static AssociativeArrayTests* s_test;
+static bool s_failNonblockingAlloc = false;
+
+struct TestArrayElement
+{
+    uint32_t key;
+    uint32_t value;
+};
+
+static bool TestArrayElementKeyEquals(const uint32_t& key, const TestArrayElement& item)
+{
+    return item.key == key;
+}
+
+struct MemoryHeader;
+typedef std::list<MemoryHeader*> AllocationList;
+
+struct MemoryHeader
+{
+    uint32_t size;
+    AllocationList::iterator listEntry;
+    alignas(16) uint8_t allocatedBytes[];
+};
+static_assert(offsetof(MemoryHeader, allocatedBytes) == sizeof(MemoryHeader), "Bytes should be at end of struct");
+
+static AllocationList s_allocationList;
+
+void* Memory_Alloc(uint32_t size)
```

Comment:
What keeps the tests from finding the implementation of `Memory_Alloc` defined in Memory.cpp?  Is that source file not compiled as part of building `AssociativeArrayTests`?

Reply:
Yeah, that file isn't linked; neither is most of the kext code - most of the `#includes` will fail, so you need complicated ifdefs.